components/brave_new_tab_ui/containers/newTab/index.tsx

```diff
@@ -60,6 +88,20 @@ class NewTabPage extends React.Component<Props, State> {
     }
   }
 
+  refreshActions = () => {
+    this.props.actions.getGrants()
+
+    chrome.braveRewards.getAdsEstimatedEarnings((amount: number) => {
+      this.props.actions.onAdsEstimatedEarnings(amount)
+    })
+    chrome.braveRewards.getBalanceReports((data: Record<string, NewTab.RewardsReport>) => {
+      this.props.actions.onBalanceReports(data)
+    })
+    chrome.braveRewards.fetchBalance((balance: NewTab.RewardsBalance) => {
+      this.props.actions.onBalance(balance)
+    })
```

Comment:
Can we combine these to a single action (e.g. `onRefreshDataReceived`) so we're not spamming the reducer and causing multiple action -> reducer -> render cycles unneccessarily. It looks like this is the only place these actions are called so simplifying would be ok. I guess if you wanted to keep separate and not repeat then you could use [redux's batch function](https://redux.js.org/faq/performance#how-can-i-reduce-the-number-of-store-update-events).

Reply:
this has been moved to a single action